pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -182,45 +200,72 @@ func makeSimpleImportSpans(
 						break
 					}
 					f := it.Value()
-					fspan := endKeyInclusiveSpan(f.Span)
-					if sp := span.Intersect(fspan); sp.Valid() {
-						fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
-						if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
-							fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
-						}
+					fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
+					if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
+						fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
+					}
 
-						// Lookup the size of the file being added; if the backup didn't
-						// record a file size, just assume it is 16mb for estimating.
-						sz := f.EntryCounts.DataSize
-						if sz == 0 {
-							sz = 16 << 20
+					// Lookup the size of the file being added; if the backup didn't
+					// record a file size, just assume it is 16mb for estimating.
+					sz := f.EntryCounts.DataSize
+					if sz == 0 {
+						sz = 16 << 20
+					}
+
+					if inclusiveOverlap(span, f.Span) {
+						var startKey roachpb.Key
+						var endKey roachpb.Key
+						if sp := span.Intersect(f.Span); sp.Valid() {
+							startKey = sp.Key
+							endKey = sp.EndKey
+						} else {
+							startKey = f.Span.EndKey
```

Comment:
This else and why these two assignments is -- both from to End -- could use a comment.

Reply:
I refactored this into a separate `getIntersectingFileSpan` and added some comments to that